### ISSUE TITLE
APIPhase does not report progress (and is not cancellable)

### ISSUE THREAD

**Issue by dragos** [sbt/sbt#1569]
`CompileProgress` allows clients to report progress and _cancel_ a running compilation. The Scala compiler refreshes progress on every compilation unit/phase transition, and the logic is implemented in `GlobalPhase`. Unfortunately, `APIPhase` (which is slow enough, tens of seconds on my project) does not extend `GlobalPhase`, but directly `Phase`. The end result is that when users press the cancel button, if the build has reached the API extraction phase, the whole UI blocks until the whole project extraction phase is finished. That can easily take more than 10 seconds.

The fix would be either to extend `GlobalPhase`, or if that's not possible... simply call `advanceUnit` and check `!cancelled` in its implementation. The logic in `GlobalPhase` is

```
      try {
        currentRun.currentUnit = unit
        if (!cancelled(unit)) {
          currentRun.informUnitStarting(this, unit)
          apply(unit)
        }
        currentRun.advanceUnit()
      } finally ...
```

This would greatly improve the experience in IDEs.

**Comment by jvican** [comment]
This is already fixed, latest Zinc 1.0 is using `GlobalPhase`.

**Comment by jvican** [comment]
Can someone close this ticket?
